python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -318,9 +313,7 @@ def _update(self):
                                        " Failed to update node."
                                        " Node has already been terminated.")
                 if nodes_to_terminate:
-                    self.prom_metrics.stopped_nodes.inc(
-                        len(nodes_to_terminate))
-                    self.provider.terminate_nodes(nodes_to_terminate)
+                    self._terminate_nodes_and_cleanup(nodes_to_terminate)
```

Comment:
Is it okay that we don't previously untrack nodes here?

Reply:
Untrack has an if block that makes it okay to untrack twice.

Reply:
Great :)